### PR TITLE
[feat]: [CDS-56678]: updating Helm chart list versions endpoints and logic

### PR DESCRIPTION
**Pull request by markoharness**
## Describe your changes
Endpoints that serve list of Helm chart versions needed to be update so that we support both fixed values and expressions/runtime values.

endpoint **/manifests/helm/chart/version** will be used for fixed configuration (when configuring standalone service or within pipeline configuration)

endpoint **/manifests/helm/v2/chart/version** will be used for pipeline runtime and in this case  one or more of prerequisite fields (connectorRef, chartName, bucket, region, etc...) can be set as expression/runtime as well as fixed value.

## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>

You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger utAll`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/45564)

### CODE REVIEW

**Comment by markoharness** [comment]
trigger feature-build

File: 120-ng-manager/src/main/java/io/harness/ng/core/manifests/resources/HelmChartVersionResource.java [hunk 1]
```diff
@@ -64,14 +64,14 @@ public ResponseDTO<HelmChartResponseDTO> getHelmChartVersionDetails(
...
-      @NotNull @QueryParam("fqnPath") String fqnPath, @QueryParam("connectorRef") String connectorIdentifier,
-      @QueryParam("chartName") String chartName, @QueryParam("region") String region,
+      @NotNull @QueryParam("connectorRef") String connectorIdentifier,
+      @NotNull @QueryParam("chartName") String chartName, @QueryParam("region") String region,
       @QueryParam("bucketName") String bucketName, @QueryParam("folderPath") String folderPath,
-      @QueryParam("lastTag") String lastTag) {
+      @QueryParam("lastTag") String lastTag, @NotNull @QueryParam("storeType") String storeType,
+      @QueryParam("helmVersion") String helmVersion) {
     HelmChartResponseDTO helmChartResponseDTO =
-        helmChartService.getHelmChartVersionDetails(accountId, orgIdentifier, projectIdentifier, serviceRef, fqnPath,
```
Comment (abosii-harness) [inline 120-ng-manager/src/main/java/io/harness/ng/core/manifests/resources/HelmChartVersionResource.java:73]: Shouldn't we keep old flow as well here for backward compatibility? What if we release BE without UI changes? Will not fetch charts for runtime screen fail?

**Review by abosii-harness — CHANGES_REQUESTED** [review]

**Comment by markoharness** [comment]
*[`120-ng-manager/src/main/java/io/harness/ng/core/manifests/resources/HelmChartVersionResource.java` line 0 at r1](https://reviewable.io/reviews/harness/harness-core/45564#gh-1147236718:-NRJ0aHM3j1fu0lHRgJg:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/ad2a7ae06aa42c3c8880cd0a694d93b2cd0887ea/120-ng-manager/src/main/java/io/harness/ng/core/manifests/resources/HelmChartVersionResource.java#L0)):*
<details><summary><i>Previously, abosii-harness (Alexandru Bosii) wrote…</i></summary><blockquote>

*(Reviewable was [unable](https://github.com/Reviewable/Reviewable/issues/14) to map this GitHub inline comment thread to the right spot — sorry!)*

Shouldn't we keep old flow as well here for backward compatibility? What if we release BE without UI changes? Will not fetch charts for runtime screen fail?
</blockquote></details>

Done.

**Comment by markoharness** [comment]
trigger feature-build

**Review by abosii-harness — DISMISSED** [review]
> Reviewed 2 of 3 files at r1, 2 of 2 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/45564)* status: 4 file(s) to review. 1 of 2 approvals obtained.

**Comment by markoharness** [comment]
trigger feature-build

**Review by abosii-harness — DISMISSED** [review]
> Reviewed 1 of 1 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/45564)* status: 4 file(s) to review. 1 of 2 approvals obtained.

**Comment by markoharness** [comment]
trigger feature-build

**Comment by markoharness** [comment]
trigger utAll

**Review by abosii-harness — APPROVED** [review]
> Reviewed 2 of 2 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/45564)* status: 4 file(s) to review. 1 of 2 approvals obtained.

**Review by abhinavsingh1196 — APPROVED** [review]
> Reviewed 1 of 3 files at r1, 1 of 2 files at r2, 2 of 2 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/45564)* status: :shipit: complete! ✓

**Pull request merged**